helpers/utils.py

```diff
@@ -175,70 +215,126 @@ def setup_kubernetes_secret_literal(username: str, namespace: str, k8s_context:
 
+
 def get_management_label(label: bool = True) -> str:
-    return 'app.kubernetes.io/managed-by=spark-client' if label else 'app.kubernetes.io/managed-by'
+    return (
+        "app.kubernetes.io/managed-by=spark-client"
+        if label
+        else "app.kubernetes.io/managed-by"
+    )
+
 
 def get_primary_label(label: bool = True) -> str:
-    return 'app.kubernetes.io/spark-client-primary=1' if label else 'app.kubernetes.io/spark-client-primary'
+    return (
+        "app.kubernetes.io/spark-client-primary=1"
+        if label
+        else "app.kubernetes.io/spark-client-primary"
+    )
+
 
 def retrieve_k8s_resources_by_label(namespace: str, k8s_context: str) -> Dict:
     kubectl_cmd = build_kubectl_cmd(namespace, k8s_context)
     label = get_management_label()
     conf = dict()
 
     cmd = f"{kubectl_cmd} get serviceaccount -l {label} -o jsonpath={{.items[0].metadata.name}}"
-    conf['spark.kubernetes.authenticate.driver.serviceAccountName'] = execute_kubectl_cmd(cmd, EXIT_CODE_GET_K8S_PROPS_FAILED)
+    conf[
+        "spark.kubernetes.authenticate.driver.serviceAccountName"
+    ] = execute_kubectl_cmd(cmd, EXIT_CODE_GET_K8S_PROPS_FAILED)
 
     cmd = f"{kubectl_cmd} get rolebinding -l {label} -o jsonpath={{.items[0].metadata.name}}"
-    conf['spark-client.rolebinding'] = execute_kubectl_cmd(cmd, EXIT_CODE_GET_K8S_PROPS_FAILED)
+    conf["spark-client.rolebinding"] = execute_kubectl_cmd(
+        cmd, EXIT_CODE_GET_K8S_PROPS_FAILED
+    )
 
     return conf
 
+
 def retrieve_primary_service_account_details() -> Dict:
     kubeconfig = get_kube_config()
     kubectl_cmd = get_kubectl_cmd()
     label = get_primary_label()
     cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig}  get serviceaccount -l {label} -A -o yaml"
     out_yaml_str = execute_kubectl_cmd(cmd, EXIT_CODE_GET_PRIMARY_RESOURCES_FAILED)
+    if out_yaml_str is None:
+        raise ValueError("could not get the secret")
     out = yaml.safe_load(out_yaml_str)
     result = dict()
-    if len(out['items']) > 0:
-        result['spark.kubernetes.authenticate.driver.serviceAccountName'] = out['items'][0]['metadata']['name']
-        result['spark.kubernetes.namespace'] = out['items'][0]['metadata']['namespace']
+    if len(out["items"]) > 0:
+        result["spark.kubernetes.authenticate.driver.serviceAccountName"] = out[
+            "items"
+        ][0]["metadata"]["name"]
+        result["spark.kubernetes.namespace"] = out["items"][0]["metadata"]["namespace"]
     return result
 
+
 def is_primary_sa_defined() -> bool:
     conf = retrieve_primary_service_account_details()
     return len(conf.keys()) > 0
+
+
+def mkdir(path: PathLike) -> None:
+    """
+    Create a dir, using a formulation consistent between 2.x and 3.x python versions.
+    :param path: path to create
+    :raises OSError: whenever OSError is raised by makedirs and it's not because the directory exists
+    """
+    try:
+        os.makedirs(path)
```

Comment:
Thanks for adding these functions, were these added automatically?  There might be more such utilities with different behaviors in 2.x and 3.x. Let's discuss.

Reply:
Copied and paste from a repo I had. To be honest, given it is 2022 (almost 2023) and we don't have the need to support backward compatibility to 2.x, I would really just drop support for older version of python. I'd stick to ">3.9" (I have updated updated CI to reflect this), but I don't have super strong opinion on the actual version we want to start comply with. Just 2.x feels very very very very old... (any strong opinion by @delgod or @taurus-forever would be much appreciated)

I believe it would be eventually good to have core utilities function, if we see the use case for it, to promote patterns, best-practices, and so, even in a form of a python package where we centralize these code. But I suppose I'd wait a bit to understand if we have such use-cases. So far, I just rely on copy and paste ;)